app/controllers/physical_ticket_controller.rb

```diff
@@ -0,0 +1,29 @@
+class PhysicalTicketController < ApplicationController
+  before_filter :authenticate_user!, except: [:new, :create]
+  load_resource :conference, find_by: :short_title
+  load_and_authorize_resource
+  authorize_resource :conference_registrations, class: Registration
+
+  def index
+    @physical_tickets = @conference.physical_tickets.by_user(current_user)
+  end
+
+  def new
+    @current_ticket_purchases = current_user.current_ticket_purchases(@conference)
+  end
+
+  def show
+  end
+
+  def create
+    @ticket_purchases = current_user.ticket_purchases.by_conference(@conference)
+    @ticket_purchases.each do |ticket_purchase|
+      if !ticket_purchase.contains_physical_tickets?
```

Comment:
Sorry my bad did you mean we should create tickets in after_create and after_update callbacks of ticket purchase model?